src/robotFactory.js

```diff
@@ -1,10 +1,81 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, {
+    x = 0,
+    y = 0,
+  }, chipVersion) {
+    Object.assign(this, {
+      name,
+      weight,
+      coords: {
+        x,
+        y,
+      },
+      chipVersion,
+    });
+  }
 
-class FlyingRobot {}
+  getInfo() {
+    return `Robot: ${this.name}, Chip version: ${this.chipVersion},`
+    + ` Weight: ${this.weight}`;
+  }
 
-class DeliveryDrone {}
+  goForward(steps = 1) {
+    this.coords.y += steps;
+  }
+
+  goBack(steps = 1) {
+    this.coords.y -= steps;
+  }
+
+  goRight(steps = 1) {
+    this.coords.x += steps;
+  }
+
+  goLeft(steps = 1) {
+    this.coords.x -= steps;
+  }
+}
+
+class FlyingRobot extends BaseRobot {
+  constructor() {
+    super(...arguments);
+
+    if (!this.coords.z) {
+      this.coords.z = arguments[2].z || 0;
+    }
+  }
+
+  goUp(steps = 1) {
+    this.coords.z += steps;
+  }
+
+  goDown(steps = 1) {
+    this.coords.z -= steps;
+  }
+}
+
+class DeliveryDrone extends FlyingRobot {
+  constructor() {
+    super(...arguments);
+
+    Object.assign(this, {
+      maxLoadWeight: arguments[4],
+      currentLoad: arguments[5] || null,
+    });
+  }
```

Comment:
I think it would be more readable if you write every argument name instead of using `arguments[4]`
```suggestion
   constructor(name, weight, coords, chipVersion, maxLoadWeight, currentLoad) {
    super(name, weight, coords, chipVersion);
    this.maxLoadWeight = maxLoadWeight;
    this.currentLoad = currentLoad || null;
  }
```

Reply:
Changed in that class and in FlyingRobot too :)